observation/message_writer.go

```diff
@@ -0,0 +1,58 @@
+package observation
+
+import (
+	"github.com/ONSdigital/dp-observation-extractor/schema"
+	"github.com/ONSdigital/go-ns/log"
+)
+
+// MessageWriter writes observations as messages
+type MessageWriter struct {
+	messageProducer MessageProducer
+}
+
+// MessageProducer dependency that writes messages
+type MessageProducer interface {
+	Output() chan []byte
+	Closer() chan bool
+}
+
+// NewMessageWriter returns a new observation message writer.
+func NewMessageWriter(messageProducer MessageProducer) *MessageWriter {
+	return &MessageWriter{
+		messageProducer: messageProducer,
+	}
+}
+
+// WriteAll observations as messages from the given observation reader.
+func (messageWriter MessageWriter) WriteAll(reader Reader, instanceID string) error {
```

Comment:
Good spot. I added it initially thinking there would be a case if Kafka fails there is no point continuing the loop. I need to discuss this further with the Kafka experts to see if there is a way to identify this. I have removed the error for now as it serves no purpose.